lib/stack_master/parameter_resolvers/latest_container.rb

```diff
@@ -0,0 +1,49 @@
+module StackMaster
+  module ParameterResolvers
+    class LatestContainer < Resolver
+      array_resolver class_name: 'LatestContainers'
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+      end
+
+      def resolve(parameters)
+        @region = parameters['region'] || @stack_definition.region
+        ecr_client = Aws::ECR::Client.new(region: @region)
+        if parameters['repository_name'].nil?
+          raise ArgumentError, "repository_name parameter is required but was not supplied"
+        end
+        images = fetch_images(parameters['repository_name'], parameters['registry_id'], ecr_client)
+        return nil if images.empty?
+        images.sort! { |image_x, image_y| image_y.image_pushed_at <=> image_x.image_pushed_at }
+        latest_image = images.first
+        latest_tag = latest_image.image_tags.delete_if { |tag| tag == "latest" }.first
+        # aws_account_id.dkr.ecr.region.amazonaws.com
+        return  "#{latest_image.registry_id}.dkr.ecr.#{@region}.amazonaws.com/#{parameters['repository_name']}:#{latest_tag}"
```

Comment:
Is the tag safe to use here? Wouldn't we be better with `image_digest`?

Reply:
Hmm, recent events have made me question this view.
It seems as though we use our repositories not just to store the latest images but also as a cache for our CI pipelines. It's quite common for pipelines to push up images from a test environment. So finding the latest image is probably a terrible idea and what we should be doing instead is tagging latest, but fetching the sha for latest and returning that.